rust/src/types/stack.rs

```diff
@@ -4,6 +4,12 @@ use std::sync::Mutex;
 
 use crate::types::{u256, FailStatus};
 
+struct NonZero<const N: usize>;
+
+impl<const N: usize> NonZero<N> {
+    const VALID: () = assert!(N > 0);
```

Comment:
is this the reason for the new toolchain support?

Reply:
No, the inline consts which have now been removed. But it is needed for #965 anyway.